contracts/vaults/Base.sol

```diff
@@ -201,171 +126,157 @@ contract Base is Ownable, ReentrancyGuard {
-        if (arbitrator != msg.sender) revert OnlyArbitrator();
+        if (registry.arbitrator() != msg.sender) revert OnlyArbitrator();
         _;
     }
 
     modifier noSafetyPeriod() {
-        //disable withdraw for safetyPeriod (e.g 1 hour) after each withdrawPeriod(e.g 11 hours)
+        HATVaultsRegistry.GeneralParameters memory generalParameters = registry.getGeneralParameters();
+        // disable withdraw for safetyPeriod (e.g 1 hour) after each withdrawPeriod(e.g 11 hours)
         // solhint-disable-next-line not-rely-on-time
         if (block.timestamp %
         (generalParameters.withdrawPeriod + generalParameters.safetyPeriod) >=
             generalParameters.withdrawPeriod) revert SafetyPeriod();
         _;
     }
 
-    modifier noActiveClaims(uint256 _pid) {
-        if (activeClaims[_pid] != 0) revert ActiveClaimExists();
+    modifier noActiveClaim() {
+        if (activeClaim.createdAt != 0) revert ActiveClaimExists();
+        _;
+    }
+
+    modifier activeClaimExists() {
+        if (activeClaim.createdAt == 0) revert NoActiveClaimExists();
         _;
     }
 
+    function initialize(
+        IRewardController _rewardController,
+        uint256 _vestingDuration,
+        uint256 _vestingPeriods,
+        uint256 _maxBounty,
+        BountySplit memory _bountySplit,
+        IERC20 _asset,
+        address _committee,
+        bool _isPaused
+    ) external initializer {
+        if (_vestingDuration > 120 days)
+            revert VestingDurationTooLong();
+        if (_vestingPeriods == 0) revert VestingPeriodsCannotBeZero();
+        if (_vestingDuration < _vestingPeriods)
+            revert VestingDurationSmallerThanPeriods();
+        if (_committee == address(0)) revert CommitteeIsZero();
+        if (address(_asset) == address(0)) revert AssetIsZero();
+        if (_maxBounty > HUNDRED_PERCENT)
+            revert MaxBountyCannotBeMoreThanHundredPercent();
+        validateSplit(_bountySplit);
+        __ERC4626_init(IERC20MetadataUpgradeable(address(_asset)));
+        rewardController = _rewardController;
+        vestingDuration = _vestingDuration;
+        vestingPeriods = _vestingPeriods;
+        maxBounty = _maxBounty;
+        bountySplit = _bountySplit;
+        committee = _committee;
+        depositPause = _isPaused;
+        HATVaultsRegistry _registry = HATVaultsRegistry(msg.sender);
+        registry = _registry;
+        __ReentrancyGuard_init();
+        _transferOwnership(_registry.owner());
+        tokenLockFactory = _registry.tokenLockFactory();
+    }
+
+    /**
+    * @notice change the information of the vault
+    * ony calleable by the owner of the contract
+    * @param _visible is this vault visible in the UI
+    * @param _depositPause pause deposits (default false).
+    * This parameter can be used by the UI to include or exclude the vault
+    * @param _descriptionHash the hash of the vault's description.
+    */
+    function updateVaultInfo(
```

Comment:
move to params.sol

Reply:
I just broke it down here https://github.com/hats-finance/hats-contracts/pull/281/commits/ed47dd31cc4644516bda6fbc58c12511e9ecd94d